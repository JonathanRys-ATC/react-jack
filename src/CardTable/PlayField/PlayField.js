import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Hand from './Hand/Hand'

import { connect } from 'react-redux'

const mapStateToProps = function (state) {
    return {
        hands: state.player.hands
    }
}

export class PlayField extends Component {
    shouldComponentUpdate(nextProps) {
        if (this.props.hands === nextProps.hands) return false;
        return true;
    }
    render() {
        return (
            <div>
                {this.props.hands.map((hand, index) => <Hand key={`hand-${index}`} hand={hand} faceUp={index} />)}
            </div>
        )
    }
}

PlayField.propTypes = {
    hands: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(PlayField)
