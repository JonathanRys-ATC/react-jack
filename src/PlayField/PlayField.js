import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Dealer from './Dealer/Dealer'
import Deck from './Deck/Deck'
import Player from './Player/Player'

import { connect } from 'react-redux'

const mapStateToProps = (state) => {
    return {
        hands: state.player.hands
    }
}

export class PlayField extends Component {
    render() {
        return (
            <div>
                <Dealer />
                <Deck />
                {this.props.hands.map((hand, index) => <Player key={`hand-${index}`} hand={hand} faceUp={index} />)}
            </div>
        )
    }
}

PlayField.propTypes = {
    hands: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(PlayField)
