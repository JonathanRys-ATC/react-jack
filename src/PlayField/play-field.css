.PlayField_main {
    border: 3px groove #aaa;
    border-radius: 10px;
    margin-top: 20px;
    padding-top: 10px;
}