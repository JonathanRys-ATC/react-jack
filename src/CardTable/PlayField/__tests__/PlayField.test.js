import React from 'react'
import { shallow } from 'enzyme'

import { PlayField } from '../PlayField'

const mockProps = { hands: [["SA", "CK"], ["D2", "HJ"]] }

describe("Test PlayField component", () => {
    it("matches the snapshot", () => {
        const testProps = { ...mockProps }
        const component = shallow(<PlayField {...testProps} />)
        expect(component).toMatchSnapshot()
    })

    it("should only render when the props have changed", () => {
        const testProps = { ...mockProps }
        const component = shallow(<PlayField {...testProps} />)
        const shouldntUpdate = component.instance().shouldComponentUpdate({ ...testProps })
        expect(shouldntUpdate).toBe(false)
        // negative test
        testProps.hands = [["H9", "D5"], ["CA", "HQ"]]
        const shouldUpdate = component.instance().shouldComponentUpdate({ ...testProps })
        expect(shouldUpdate).toBe(true)
    })
})