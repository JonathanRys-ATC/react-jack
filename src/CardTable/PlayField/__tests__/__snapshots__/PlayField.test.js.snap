// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlayField component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayField
    hands={
      Array [
        Array [
          "SA",
          "CK",
        ],
        Array [
          "D2",
          "HJ",
        ],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Hand
          faceUp={0}
          hand={
            Array [
              "SA",
              "CK",
            ]
          }
        />,
        <Hand
          faceUp={1}
          hand={
            Array [
              "D2",
              "HJ",
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "hand-0",
        "nodeType": "class",
        "props": Object {
          "faceUp": 0,
          "hand": Array [
            "SA",
            "CK",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "hand-1",
        "nodeType": "class",
        "props": Object {
          "faceUp": 1,
          "hand": Array [
            "D2",
            "HJ",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Hand
            faceUp={0}
            hand={
              Array [
                "SA",
                "CK",
              ]
            }
          />,
          <Hand
            faceUp={1}
            hand={
              Array [
                "D2",
                "HJ",
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "hand-0",
          "nodeType": "class",
          "props": Object {
            "faceUp": 0,
            "hand": Array [
              "SA",
              "CK",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "hand-1",
          "nodeType": "class",
          "props": Object {
            "faceUp": 1,
            "hand": Array [
              "D2",
              "HJ",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
